src/core/server/saved_objects/service/lib/search_dsl/query_params.ts

```diff
@@ -129,6 +129,7 @@ interface QueryParams {
   registry: ISavedObjectTypeRegistry;
   namespaces?: string[];
   type?: string | string[];
+  typesAndNamespacesMap?: Map<string, string[] | undefined>;
```

Comment:
> I would avoid doing that on the SOC/SOR public API, as it would introduce unnecessary complexity for the consumers (especially on OSS). However changing the internal helpers (getSearchDsl, getQueryParams) would probably make sense. Still probably minor though, so I would keep it as optional, feel free to ignore.

My thoughts exactly! I don't think it makes sense to rework the public API here